lib/plugins/splitIOFeatureManagerPlugin.ts

```diff
@@ -0,0 +1,123 @@
+import * as path from 'path'
+import * as process from 'process'
+
+import { SplitFactory } from '@splitsoftware/splitio'
+import type SplitIO from '@splitsoftware/splitio/types/splitio'
+import type {
+  Attributes,
+  Properties,
+  SplitKey,
+  Treatment,
+  TreatmentWithConfig,
+} from '@splitsoftware/splitio/types/splitio'
+import type { FastifyInstance } from 'fastify'
+import fp from 'fastify-plugin'
+
+const DISABLED_TREATMENT: Treatment = 'control'
+
+declare module 'fastify' {
+  interface FastifyInstance {
+    splitIOFeatureManager: SplitIOFeatureManager
+  }
+}
+
+export interface SplitIOOptions {
+  isEnabled: boolean
+  apiKey: string
+  debugMode: boolean
+  localhostFilePath?: string
+}
+
+export class SplitIOFeatureManager {
+  private readonly isEnabled: boolean
+
+  private readonly splitIOClient?: SplitIO.IClient
+
+  constructor(
+    isSplitIOEnabled: boolean,
+    apiKey: string,
+    debug: boolean,
+    localhostFilePath?: string,
+  ) {
+    if (isSplitIOEnabled) {
+      const factory: SplitIO.ISDK = SplitFactory({
+        core: {
+          authorizationKey: localhostFilePath ? 'localhost' : apiKey,
+        },
+        features: localhostFilePath ? path.join(process.cwd(), localhostFilePath) : undefined,
```

Comment:
I guess there's a chance `process.cwd()` might not yield correct result outside localhost? Depends on how `node` is triggered, but will require testing I suppose.